### ISSUE TITLE
v1.31.3+k3s1 Upgrade Images missing

### ISSUE THREAD

**Issue by Xeroxxx** [k3s-io/k3s#11416]
**Environmental Info:**
K3s Version: v1.31.2+k3s1

Node(s) CPU architecture, OS, and Version: Linux 6.11.10-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.11.10-1 (2024-11-23) x86_64 GNU/Linux

Cluster Configuration: 1 server, 2 agents

**Describe the bug:**

System-Upgrade Controller fails to pull however it cordons the server node.
"docker.io/rancher/k3s-upgrade:v1.31.3-k3s1": docker.io/rancher/k3s-upgrade:v1.31.3-k3s1: not found

Release channel is set to latest.

**Steps To Reproduce:**

Install system-upgrade controller on v1.31.2+k3s1 with release channel latest

**Expected behavior:**

- Do not upgrade or cordon nodes if images are missing
- Or images are present

**Actual behavior:**
Cordon server nodes, not finding the images, jobs are stuck

**Additional context / logs:**

**Comment by janpieper** [comment]
In https://github.com/rancher/system-upgrade-controller/issues/339 @brandond wrote that there was a problem with the CI. The issue is known and worked on.

**Comment by janpieper** [comment]
The images [k3s-upgrade:v1.31.3-k3s1](https://hub.docker.com/r/rancher/k3s-upgrade/tags?name=v1.31.3-k3s1) are finally available :+1:

**Comment by Xeroxxx** [comment]
Yes, thank you. My cluster is already running 1.31.3 as system-upgrade-controller basically hammers the registry :D
